function yayOrNay() {
  return new Promise((resolve, reject) => {
    const val = Math.round(Math.random() * 1); // 0 or 1, at random

    val ? resolve('Lucky!!') : reject('Nope ğŸ˜ ');
  });
}

async function msg() {
  try {
    const msg = await yayOrNay();
    console.log(msg);
  } catch(err) {
    console.log(err);
  }
  // o cÃ³digo que estiver aqui Ã© executado mesmo se der erro.
}

// a saÃ­da varia porque usa Math.random para retornar sucesso ou falha.
msg(); // Lucky!!
msg(); // Lucky!!
msg(); // Lucky!!
msg(); // Nope ğŸ˜ 
msg(); // Lucky!!
msg(); // Nope ğŸ˜ 
msg(); // Nope ğŸ˜ 
msg(); // Nope ğŸ˜ 
msg(); // Nope ğŸ˜ 
msg(); // Lucky!!